go/ephemeral/user_ek.go

```diff
@@ -302,24 +344,12 @@ func getLatestUserEKMetadataMaybeStale(ctx context.Context, g *libkb.GlobalConte
 func getActiveUserEKMetadata(ctx context.Context, g *libkb.GlobalContext, merkleRoot libkb.MerkleRoot) (metadata *keybase1.UserEkMetadata, err error) {
 	defer g.CTrace(ctx, "GetActiveUserEKMetadata", func() error { return err })()
 
-	metadata, err = getLatestUserEKMetadataMaybeStale(ctx, g)
-	if err != nil {
-		return nil, err
-	}
-	if metadata == nil {
-		// There is no key.
-		return nil, nil
-	}
+	currentMerkleRoot, err := g.GetMerkleClient().FetchRootFromServer(ctx, libkb.EphemeralKeyMerkleFreshness)
 
-	// Check whether the key is stale. This isn't considered an error,
-	// since the server doesn't do this check for us. We log these cases
-	// and return nil.
-	ageSecs := merkleRoot.Ctime() - metadata.Ctime.UnixSeconds()
-	if ageSecs > KeyLifetimeSecs {
-		g.Log.CDebugf(ctx, "found stale userEK %s", metadata.Kid)
-		return nil, nil
+	allActive, err := getAllActiveUserEKs(ctx, g, keybase1.TimeFromSeconds(currentMerkleRoot.Ctime()))
+	if err != nil || len(allActive) == 0 {
+		return nil, err
 	}
 
-	// This key is valid and current. Return it.
-	return metadata, nil
+	return &allActive[len(allActive)-1], nil
```

Comment:
i feel it's better to just call max here instead of depending on it being sorted/we stuff the latest at the end

Reply:
or really just call `maybeStale` and do the staleness check on the single key so we don't have iterate?